spec/researches/buildKeywordFormsSpec.js

```diff
@@ -1,89 +1,204 @@
-		expect( result ).toContain( "keywords'" );
-		expect( result ).toContain( "keywords`" );
-		expect( result ).toContain( "keywords‘" );
-		expect( result ).toContain( "keywords‛" );
-		expect( result ).toContain( "keywords’s" );
-		expect( result ).toContain( "keywords's" );
-		expect( result ).toContain( "keywords`s" );
-		expect( result ).toContain( "keywords‘s" );
-		expect( result ).toContain( "keywords‛s" );
-
-		mockPaper = new Paper( "", { keyword: "Keyword", locale: "en_EN" } );
-		result = buildKeywordForms( mockPaper );
-		expect( result ).toContain( "keyword" );
-		expect( result ).toContain( "keywords" );
-
-
-		mockPaper = new Paper( "", { keyword: "\"keyword\"", locale: "en_EN" } );
-		result = buildKeywordForms( mockPaper );
-		expect( result ).toEqual( [].concat( "keyword" ) );
-
-		mockPaper = new Paper( "", { keyword: "“keyword”", locale: "en_EN" } );
-		result = buildKeywordForms( mockPaper );
-		expect( result ).toEqual( [].concat( "keyword" ) );
-
-		mockPaper = new Paper( "", { keyword: "\"keyword", locale: "en_EN" } );
-		result = buildKeywordForms( mockPaper );
-		expect( result ).toContain( "\"keyword" );
-		expect( result ).toContain( "\"keywords" );
-
-		mockPaper = new Paper( "", { keyword: "\"Keyword\"", locale: "en_EN" } );
-		result = buildKeywordForms( mockPaper );
-		expect( result ).toEqual( [].concat( "Keyword" ) );
-
-		mockPaper = new Paper( "", { keyword: "\"keyword and keyphrases\"", locale: "en_EN" } );
-		result = buildKeywordForms( mockPaper );
-		expect( result ).toEqual( [].concat( "keyword and keyphrases" ) );
-
-		mockPaper = new Paper( "", { keyword: "keyword and keyphrases", locale: "en_EN" } );
-		result = buildKeywordForms( mockPaper );
-		expect( result ).toContain( "keyword" );
-		expect( result ).toContain( "keywords" );
-		expect( result ).toContain( "keyphrase" );
-		expect( result ).toContain( "keyphrases" );
-		const whetherResultContainsWord = includes( result, "and" );
-		expect( whetherResultContainsWord ).toBe( false );
-
-		mockPaper = new Paper( "", { keyword: "One and two", locale: "en_EN" } );
-		result = buildKeywordForms( mockPaper );
-		expect( result ).toContain( "one" );
-		expect( result ).toContain( "ones" );
-		expect( result ).toContain( "and" );
-		expect( result ).toContain( "ands" );
-		expect( result ).toContain( "two" );
-		expect( result ).toContain( "twos" );
+describe( "A test for building keyword and synonyms forms for a paper", function() {
+	it( "returns the exact matches if the input strings are embedded in quotation marks and word forms if not; for empty locale", function() {
+		const keyword = "\"I am going for a walk\"";
+		const synonyms = "\"You are not going for a walk\", You are going for a movie, And he is going to work diligently.";
+		let locale;
+
+		const expectedResult = {
+			keyphraseForms: [ [ "I am going for a walk" ] ],
+			synonymsForms: [
+				[ [ "You are not going for a walk" ] ],
+				[ movieForms ],
+				[ workForms, diligentlyForms ],
```

Comment:
Really nice structure with these sample sentences and sample forms to keep the specs clean and easy to read!